src/dune_console/progress.ml

```diff
@@ -3,25 +3,44 @@ open Stdune
 module No_flush = struct
   let status_line = ref Pp.nop
   let start () = ()
-  let status_line_len = ref 0
+  let status_line_height = ref 0
+
+  (* This ansi code will clear the entire line leaving the cursor in place. *)
+  (* TODO: should this be "\027[K" instead? *)
+  let ansi_erase = "\027[2K"
+
+  (* Move the cursor up one line. *)
+  let ansi_up = "\027M"
+  let clear_code_length = String.length ansi_erase + String.length ansi_up
 
   let hide_status_line () =
-    if !status_line_len > 0 then Printf.eprintf "\r%*s\r" !status_line_len ""
+    if !status_line_height > 0
+    then (
+      let buf = Buffer.create ((clear_code_length * !status_line_height) - 1) in
+      (* Return the cursor to the begining of the line. *)
+      Buffer.add_char buf '\r';
+      (* First we clear the line. *)
+      Buffer.add_string buf ansi_erase;
+      for _ = 1 to !status_line_height - 1 do
+        (* If we have multiple lines, we move the cursor up and clear. *)
+        Buffer.add_string buf (ansi_up ^ ansi_erase)
+      done;
+      Buffer.contents buf |> Printf.eprintf "%s")
```

Comment:
Why do you add this extra layer of buffering? You can just use print directly and it's already buffered.